crates/pg_workspace/src/workspace/server.rs

```diff
@@ -358,51 +300,51 @@ impl Workspace for WorkspaceServer {
 
         // run diagnostics for each statement in parallel if its mostly i/o work
         if let Ok(connection) = self.connection.read() {
-            if let Some(pool) = connection.get_pool() {
-                let typecheck_params: Vec<_> = doc
-                    .iter_statements_with_text_and_range()
-                    .map(|(stmt, range, text)| {
-                        let ast = self.pg_query.get_ast(&stmt);
-                        let tree = self.tree_sitter.get_parse_tree(&stmt);
-                        (text.to_string(), ast, tree, *range)
-                    })
-                    .collect();
-
-                let pool_clone = pool.clone();
-                let path_clone = params.path.clone();
-                let async_results = run_async(async move {
-                    stream::iter(typecheck_params)
-                        .map(|(text, ast, tree, range)| {
-                            let pool = pool_clone.clone();
-                            let path = path_clone.clone();
-                            async move {
-                                if let Some(ast) = ast {
-                                    pg_typecheck::check_sql(TypecheckParams {
-                                        conn: &pool,
-                                        sql: &text,
-                                        ast: &ast,
-                                        tree: tree.as_deref(),
-                                    })
-                                    .await
-                                    .map(|d| {
-                                        let r = d.location().span.map(|span| span + range.start());
-
-                                        d.with_file_path(path.as_path().display().to_string())
-                                            .with_file_span(r.unwrap_or(range))
-                                    })
-                                } else {
-                                    None
-                                }
+            let pool = connection.get_pool();
```

Comment:
all of the below is just moved by a tab, because I removed the `if let Some(..) = ..` closure